coredns/endpointslice/map.go

```diff
@@ -111,6 +122,41 @@ func (m *Map) Put(es *discovery.EndpointSlice) {
 		return
 	}
 
+	// TODO: Only do this look up if globalnet enabled to reduce unnessary call to Kube API
+	var localSlice *discovery.EndpointSlice
+	if cluster == m.localClusterID {
+		clientSet, err := kubernetes.NewForConfig(m.kubeConfig)
+		if err != nil {
+			klog.Warningf("Error creating client set %#v", err)
+			return
+		}
+
+		retryErr := retry.OnError(retry.DefaultBackoff, func(err error) bool {
+			return !apierrors.IsNotFound(err)
+		}, func() error {
+			localSlices, err := clientSet.DiscoveryV1().EndpointSlices(es.Labels[constants.LabelSourceNamespace]).List(context.TODO(), metav1.ListOptions{
+				LabelSelector: labels.Set(map[string]string{
+					constants.KubernetesServiceName: es.Labels[constants.MCSLabelServiceName],
+				}).String(),
+			})
+			if err != nil && !apierrors.IsNotFound(err) {
+				return err
+			} else if err != nil || len(localSlices.Items) != 1 {
```

Comment:
We don't really need to check for `IsNotFound` error - it would be reported on L152 below anyway. Also for the `List` operation, it doesn't return `IsNotFound` if there's no items found - it only returns `IsNotFound` if the resource type isn't found which shouldn't happen.

```suggestion
			if err != nil {
				return err
			} 
			
			if len(localSlices.Items) != 1 {
```